DifferentialGrowth2D/DifferentialPolyline.js

```diff
@@ -0,0 +1,236 @@
+import { Vector2 } from "./AcmeVector.js";
+import { DifferentialNode, NEARBY_RADIUS } from "./node.js";
+import { Circle } from "./circle.js";
+
+const MAX_EDGE_LENGTH = 150;
+
+const MAX_POINTS_PER_POLYLINE = 1000;
+
+// How small of an angle below which more nodes will be added
+const PINCH_ANGLE_THRESHOLD = Math.PI / 6;
+const PINCH_THRESHOLD = Math.cos(Math.PI / 4);
+
+const MAX_SPEED = 10;
+const MAX_FORCE = 20;
+
+// Attraction between neighboring points is modeled like springs.
+const ATTRACTION_MIN_DISTANCE = 20;
+
+// Temporary accumulators since vectors are added in-place
+const TEMP_DESIRED_VELOCITY = new Vector2();
+const TEMP_STEERING_FORCE = new Vector2();
+const TEMP_TOTAL_FORCE = new Vector2();
+const TEMP_ATTRACTION = new Vector2();
+const TEMP_REPULSION = new Vector2();
+const TEMP_BA = new Vector2();
+const TEMP_BC = new Vector2();
+
+// Because JavaScript doesn't handle the modulo operator
+// correctly for negative numbers
+function fixed_modulo(x, modulus) {
+  return ((x % modulus) + modulus) % modulus;
+}
+
+export class DifferentialPolyline {
```

Comment:
It would be nice to smooth out the curve. Consider treating the polyline as a B-spline, then converting it to Bezier curves (see [B-spline Wikipedia article](https://en.wikipedia.org/wiki/B-spline#Cubic_B-Splines))

Reply:
This worked out quite nicely!